tests/h/admin/views/mailer_test.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+
+from pyramid.httpexceptions import HTTPSeeOther
+
+from h.admin.views import mailer as views
+
+
+class TestMailerIndex(object):
+
+    def test_when_no_taskid(self, pyramid_request):
+        result = views.mailer_index(pyramid_request)
+
+        assert result == {'taskid': None}
+
+    def test_with_taskid(self, pyramid_request):
+        pyramid_request.params['taskid'] = 'abcd1234'
+
+        result = views.mailer_index(pyramid_request)
+
+        assert result == {'taskid': 'abcd1234'}
+
+
+@pytest.mark.usefixtures('mailer', 'testmail', 'routes')
+class TestMailerTest(object):
+
+    def test_doesnt_mail_when_no_recipient(self, mailer, pyramid_request):
+        views.mailer_test(pyramid_request)
+
+        assert not mailer.send.delay.called
+
+    def test_redirects_when_no_recipient(self, pyramid_request):
+        result = views.mailer_test(pyramid_request)
+
+        assert isinstance(result, HTTPSeeOther)
+        assert result.location == '/adm/mailer'
+
+    def test_sends_mail(self, mailer, pyramid_request, testmail):
+        pyramid_request.params['recipient'] = 'meerkat@example.com'
+
+        views.mailer_test(pyramid_request)
+
+        mailer.send.delay.assert_called_once_with(['meerkat@example.com'], 'TEST', 'text', 'html')
+
+    def test_redirects(self, mailer, pyramid_request, testmail):
+        pyramid_request.params['recipient'] = 'meerkat@example.com'
+
+        result = views.mailer_test(pyramid_request)
+
+        assert isinstance(result, HTTPSeeOther)
+        assert result.location == '/adm/mailer?taskid=a1b2c3'
+
+
+class FakeResult(object):
+    def __init__(self):
+        self.task_id = 'a1b2c3'
+
+
+@pytest.fixture
+def mailer(patch):
+    mailer = patch('h.admin.views.mailer.mailer')
+    mailer.send.delay.return_value = FakeResult()
+    return mailer
+
+
+@pytest.fixture
+def testmail(patch):
+    test = patch('h.admin.views.mailer.test')
+    test.generate.side_effect = lambda _, r: ([r], 'TEST', 'text', 'html')
+    return test
+
+
+@pytest.fixture
+def routes(pyramid_config):
+    pyramid_config.add_route('admin_mailer', '/adm/mailer')
```

Comment:
It's deliberate -- the view should be using the defined routes to get its URLs, not hardcoding them. Keeping the routes different in tests from in the "real" application is one way of checking that.